ui/src/app/services/conversion/conversion.service.ts

```diff
@@ -147,25 +147,47 @@ export class ConversionService {
 
   getColumnMapping(tableName: string, data: IConv): IColumnTabData[] {
     let srcTableName = data.ToSource[tableName].Name
-    return data.SrcSchema[srcTableName].ColNames.map((name: string, i: number) => {
-      let spColName = data.ToSpanner[srcTableName].Cols[name]
-      let srcPks = data.SrcSchema[srcTableName].PrimaryKeys
-      let spannerColDef = data.SpSchema[tableName].ColDefs[spColName]
-      return {
-        spOrder: spannerColDef ? i + 1 : '',
-        srcOrder: i + 1,
-        spColName: spannerColDef ? spColName : '',
-        spDataType: spannerColDef ? spannerColDef.T.Name : '',
-        srcColName: name,
-        srcDataType: data.SrcSchema[srcTableName].ColDefs[name].Type.Name,
-        spIsPk: spannerColDef
-          ? data.SpSchema[tableName].Pks.map((p) => p.Col).indexOf(spColName) !== -1
-          : false,
-        srcIsPk: srcPks ? srcPks.map((p) => p.Column).indexOf(name) !== -1 : false,
-        spIsNotNull: spannerColDef ? data.SpSchema[tableName].ColDefs[spColName].NotNull : false,
-        srcIsNotNull: data.SrcSchema[srcTableName].ColDefs[name].NotNull,
+    const res: IColumnTabData[] = data.SrcSchema[srcTableName].ColNames.map(
```

Comment:
This is part of which fix?

Reply:
@shreyakhajanchi This part is modified for adding synthetic key in spanner column.